effort_controllers/test/test_joint_group_effort_controller.cpp

```diff
@@ -58,116 +63,183 @@ void JointGroupEffortControllerTest::TearDown()
-  ASSERT_EQ(
-    test_robot_->get_joint_handle(
-      *joint2_eff_cmd_handle_), hardware_interface::hardware_interface_ret_t::OK);
-  ASSERT_EQ(
-    test_robot_->get_joint_handle(
-      *joint3_eff_cmd_handle_), hardware_interface::hardware_interface_ret_t::OK);
+  // configure failed, 'joints' parameter not set
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
 }
 
-TEST_F(JointGroupEffortControllerTest, ConfigureParamsTest)
+TEST_F(JointGroupEffortControllerTest, JointsParameterIsEmpty)
 {
-  // joint handles not initialized yet
-  ASSERT_TRUE(controller_->joint_cmd_handles_.empty());
-
   SetUpController();
-
-  // configure failed, 'joints' paremeter not set
-  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
-  controller_->lifecycle_node_->declare_parameter(
-    "joints",
-    rclcpp::ParameterValue(std::vector<std::string>()));
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>()});
 
   // configure failed, 'joints' is empty
   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
-  auto result = controller_->lifecycle_node_->set_parameter(
-    rclcpp::Parameter(
-      "joints",
-      rclcpp::ParameterValue(std::vector<std::string>{"joint1", "joint2"})));
-  ASSERT_TRUE(result.successful);
+}
+
+TEST_F(JointGroupEffortControllerTest, ConfigureAndActivateParamsSuccess)
+{
+  SetUpController();
+  controller_->get_node()->set_parameter({"joints", joint_names_});
 
   // configure successful
   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+}
+
+TEST_F(JointGroupEffortControllerTest, ActivateWithWrongJointsNamesFails)
+{
+  SetUpController();
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>{"joint1", "joint4"}});
+
+  // activate failed, 'joint4' is not a valid joint name for the hardware
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
 
-  // joint handles initialized
-  ASSERT_EQ(controller_->joint_cmd_handles_.size(), 2ul);
-  ASSERT_EQ(controller_->joint_cmd_handles_[0].get_interface_name(), "effort_command");
-  ASSERT_EQ(controller_->joint_cmd_handles_[1].get_interface_name(), "effort_command");
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>{"joint1", "joint2"}});
+
+  // activate failed, 'acceleration' is not a registered interface for `joint1`
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
 }
 
-TEST_F(JointGroupEffortControllerTest, CheckParamsTest)
+TEST_F(JointGroupEffortControllerTest, CommandSuccessTest)
```

Comment:
I am guessing you are using the same test as for the general forward controller. Is that right?